index.js

```diff
@@ -142,36 +142,32 @@ const fastifyReplyFrom = fp(function from (fastify, opts, next) {
     const requestHeaders = rewriteRequestHeaders(this.request, headers)
     const contentLength = requestHeaders['content-length']
     let requestImpl
-    if (retryMethods.has(method) && !contentLength) {
-      const retryHandler = (req, res, err, retries) => {
-        const getDefaultDelay = () => {
-          // Magic number, so why not 42? We might want to make this configurable.
-          let retryAfter = 42 * Math.random() * (retries + 1)
-
-          if (res && res.headers['retry-after']) {
-            retryAfter = res.headers['retry-after']
-          }
-          if (res && res.statusCode === 503 && req.method === 'GET') {
-            if (retriesCount === 0 && retries < maxRetriesOn503) {
-              // we should stop at some point
-              return retryAfter
-            }
-          } else if (retriesCount > retries && err && err.code === retryOnError) {
-            return retryAfter
-          }
-          return null
-        }
 
-        if (customRetry && customRetry.handler) {
-          const customRetries = customRetry.retries || 1
-          if (++retries < customRetries) {
-            return customRetry.handler({ err, req, res, getDefaultDelay })
+    const getDefaultDelay = (req, res, err, retries) => {
+      if (retryMethods.has(method) && !contentLength) {
+        // Magic number, so why not 42? We might want to make this configurable.
+        let retryAfter = 42 * Math.random() * (retries + 1)
+
+        if (res && res.headers['retry-after']) {
+          retryAfter = res.headers['retry-after']
+        }
+        if (res && res.statusCode === 503 && req.method === 'GET') {
+          if (retriesCount === 0 && retries < maxRetriesOn503) {
+            return retryAfter
           }
+        } else if (retriesCount > retries && err && err.code === retryOnError) {
+          return retryAfter
         }
-        return getDefaultDelay()
       }
+      return null
+    }
 
-      requestImpl = createRequestRetry(request, this, retryHandler)
+    if (retryDelay && retryDelay.handler) {
+      requestImpl = createRequestRetry(request, this, (req, res, err, retries) => {
+        return retryDelay.handler({ err, req, res, attempt: retries, getDefaultDelay })
+      })
+    } else if (retryMethods.has(method) && !contentLength) {
+      requestImpl = createRequestRetry(request, this, getDefaultDelay)
```

Comment:
What about just always returning it the created retry-er, instead of duplicating the methods/content length check down here?